packages/block-library/src/navigation/edit/index.js

```diff
@@ -264,54 +221,44 @@ function Navigation( {
 	const hasUnsavedBlocks = hasUncontrolledInnerBlocks && ! isEntityAvailable;
 
 	useEffect( () => {
-		if (
-			ref ||
-			! hasResolvedClassicMenus ||
-			! hasResolvedNavigationMenus ||
-			isConvertingClassicMenu ||
-			fallbackNavigationMenuId ||
-			hasUnsavedBlocks ||
-			! classicMenus?.length
-		) {
+		// If:
+		// - there is an existing menu, OR
+		// - there are existing inner blocks
+		// ...then don't request a fallback menu.
+		if ( ref || hasUnsavedBlocks ) {
 			return;
 		}
 
-		// If there's non fallback navigation menus and
-		// a classic menu with a `primary` location or slug,
-		// then create a new navigation menu based on it.
-		// Otherwise, use the most recently created classic menu.
-		const primaryMenus = classicMenus.filter(
-			( classicMenu ) =>
-				classicMenu.locations.includes( 'primary' ) ||
-				classicMenu.slug === 'primary'
-		);
-
-		if ( primaryMenus.length ) {
-			convertClassicMenu(
-				primaryMenus[ 0 ].id,
-				primaryMenus[ 0 ].name,
-				'publish'
-			);
-		} else {
-			classicMenus.sort( ( a, b ) => {
-				return b.id - a.id;
+		apiFetch( { path: '/wp/v2/navigation/fallbacks' } )
+			.then( ( fallbackNavigationMenu ) => {
+				if ( ! fallbackNavigationMenu?.ID ) {
+					showNavigationMenuStatusNotice(
+						__( 'Unable to create Navigation Menu.' )
+					);
+					return;
+				}
+
+				/**
+				 *  This fallback displays (both in editor and on front)
+				 *  a list of pages only if no menu (user assigned or
+				 *  automatically picked) is available.
+				 *  The fallback should not request a save (entity dirty state)
+				 *  nor to be undoable, hence why it is marked as non persistent
+				 */
+				__unstableMarkNextChangeAsNotPersistent();
+				setRef( fallbackNavigationMenu.ID );
+			} )
+			.catch( () => {
+				showNavigationMenuStatusNotice(
+					__( 'Failed to create Navigation Menu.' )
```

Comment:
The notice here is not informative. The error is that "We could not load a default navigation", not that we could not create one.

Reply:
✅